cmake/kokkos_arch.cmake

```diff
@@ -663,11 +663,11 @@ IF (KOKKOS_ENABLE_OPENACC)
   IF(KOKKOS_CUDA_ARCH_FLAG)
     STRING(REPLACE "sm_" "cc" NVHPC_CUDA_ARCH ${KOKKOS_CUDA_ARCH_FLAG})
     COMPILER_SPECIFIC_FLAGS(
-      NVHPC -acc -gpu=${NVHPC_CUDA_ARCH}
+      NVHPC -acc -gpu=${NVHPC_CUDA_ARCH} -DKOKKOS_ENABLE_OPENACC_COLLAPSE_HIERARCHICAL_CONSTRUCTS
```

Comment:
No what we discussed at the developer meeting was defining it in the code wherever appropriate, likely an OpenACC header file
```C++
#define KOKKOS_ENABLE_OPENACC_COLLAPSE_HIERARCHICAL_CONSTRUCTS

Reply:
I thought it would be better to add the macro definition directly to the NVHPC compiler since the collapsing transformation works only for the NVHPC compiler. 
Anyway, I moved the macro into `Kokkos_OpenACC_Team.hpp`.

Reply:
>  I moved the macro into `Kokkos_OpenACC_Team.hpp`

That's a bad idea.  Now we have a bug in `OpenACC::print_configuration`.

Reply:
Thank you for catching this bug; the macro is moved to `Kokkos_OpenACC.hpp`.